app/PermissionsContext.js

```diff
@@ -87,9 +87,13 @@ const PermissionsProvider = ({ children }) => {
     if (Platform.OS === 'ios') {
       const status = await check(PERMISSIONS.IOS.LOCATION_ALWAYS);
       setLocationPermission(statusToEnum(status));
-    } else if (Platform.OS === 'andriod') {
+    } else if (Platform.OS === 'android') {
       const status = await check(PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION);
-      setLocationPermission(statusToEnum(status));
+      if (locationPermission === PermissionStatus.UNKNOWN) {
```

Comment:
should this be checking `status`?

Why doesn't `setLocationPermission(statusToEnum(status));` work?

Reply:
@tstirrat It definitely works! The only reason I put that there was because there was initially a typo of "andriod" which meant it never entered that condition whenever the android app checked for permissions. When I fixed that, it always came out to denied initially when the android app booted which I am assuming is not the preferred behavior. So i added that if condition so that permissions for android is shown as "unknown" when the user first install the app instead of denied. I can definitely change it back to the original.

Reply:
I read this as, if the location permission is unknown, then set the location permission to unknown, regardless of what the actual permission setting being found by check is. Which i dont think is the behavior we want in all cases. I would remove the if else statement here and address the initializing as denied bug in a separate pr.

One possibility on that would be going through the onboarding flow multiple times, are you reseting the permissions on each test?

Reply:
@johnschoeman I am! I clear the data for each run I do so that I can get to the permission screens. I will remove the if and else and try to see why android is initializing it as denied

Reply:
Not sure exactly how you are clearing the data, but I believe that adb has a utility for reseting permissions: https://gist.github.com/Pulimet/5013acf2cd5b28e55036c82c91bd56d8#file-adbcommands-L73

I would give that a try and if the issue persists then you have found a bug!  and I would open an issue to capture that and we can go from there to fix it. Thanks for digging into this!

Reply:
I see! I am essentially going to the app info > storage > clear data. This resets all the permissions for me.

Reply:
So @namit-agrawal, I am also seeing that issue. The diagram below from [react-native-permissions](https://github.com/react-native-community/react-native-permissions#android-flow) should clarify why that is happening.
<img width="826" alt="Screen Shot 2020-06-02 at 9 38 28 AM" src="https://user-images.githubusercontent.com/20157849/83533162-d3a38900-a4b4-11ea-9936-e47edac47465.png">

Seems like a legitimate bug that we should address in another PR.